tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRARecoveryService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+
+/**
+ * This SPI is providing the implementation with the ability to define what actions need to be taken
+ * in case the TCK needs to wait for the the delivery of of participant callbacks (Complete, Compensate, ...) when the
+ * LRA is asked to perform some eventually consistent action {@link LRARecoveryService#waitForCallbacks(URI)} and to
+ * define what actions can be taken to make the recovery (replay of the Compensate, Complete or Status) calls happen
+ * when the TCK requires to perform such actions {@link LRARecoveryService#waitForRecovery(URI)}. This allows to not
+ * wait for the mechanisms like periodic recovery and thus it makes the TCK runs faster.
+ */
+public interface LRARecoveryService {
+
+    /**
+     * Waits for the delivery of all participant callbacks (Complete, Compensate, AfterLRA, ...). When
+     * this method returns the results are verified so the implementation must make sure that the method
+     * doesn't return before the validation can happen.
```

Comment:
I think I understand what the javadoc is saying but only because we have discussed it elsewhere on the issues.
A reader who does not have that context would struggle to understand how he/she should implement these SPI methods.